samples/net/zperf/src/main.c

```diff
@@ -41,6 +47,21 @@ static int enable_usb_device_next(void)
 
 int main(void)
 {
+
+#if defined(CONFIG_SOC_NRF5340_CPUAPP)
```

Comment:
Not a big fan of having SoC specific code directly in the sample's `main`. Maybe add a separate C file and use `SYS_INIT`. The file can then be conditionally added in CMake with `target_sources_ifdef`.

Reply:
Agreed, I did raise a Nordic internal request to provide the CPU frequency configuration support in a DTS and other abandoned avenues (https://github.com/zephyrproject-rtos/zephyr/pull/42935).

I saw this was already being used in https://github.com/zephyrproject-rtos/zephyr/blob/ee250a11ce503fdf24793afac0712bacd26759da/subsys/bluetooth/audio/shell/bap_usb.c#L554 hence used the same (just had to use `#if` to avoid undefined references).